kv_store/src/server.rs

```diff
@@ -59,19 +57,29 @@ impl Server {
 
     async fn handle_decided_entries(&mut self) {
         let new_decided_idx = self.omni_paxos.get_decided_idx();
-        if self.last_decided_idx < new_decided_idx {
-            let decided_entries = self.omni_paxos.read_decided_suffix(self.last_decided_idx).unwrap();
+        if self.last_decided_idx < new_decided_idx as u64 {
+            let decided_entries = self
+                .omni_paxos
+                .read_decided_suffix(self.last_decided_idx as usize)
+                .unwrap();
             self.update_database(decided_entries);
-            self.last_decided_idx = new_decided_idx;
+            self.last_decided_idx = new_decided_idx as u64;
```

Comment:
I think if we just make the `self.last_decided_idx` to be usize, you can avoid a bunch of conversions to u64.

Reply:
I got compilation errors without the cast but I will check again each one individually

Reply:
![image](https://github.com/user-attachments/assets/a4529a74-17b1-4c8e-975f-f67207748568)
I had this error when doing `cargo check`

Reply:
yeah so if we make self.last_decided_idx usize then we just need to do the type casting on line 68